lib/services/post.ts

```diff
@@ -0,0 +1,155 @@
+import { Post, PostStatus, User } from "@prisma/client";
+import { prisma } from "./db";
+import { deleteImage, uploadImage } from "./image";
+
+export type PostWithUser = Post & {
+  user: User;
+};
+
+export type CreatablePost = {
+  price: number;
+  description: string;
+  image: Buffer;
+  bookId: string;
+  userId: string;
+};
+
+export type UpdatablePost = {
+  price?: number;
+  description?: string;
+  image?: Buffer;
+  status?: PostStatus;
+};
+
+type UpdatablePostResolved = {
+  price?: number;
+  description?: string;
+  imageUrl?: string;
+  status?: PostStatus;
+};
+
+/**
+ * Returns the post with the given id.
+ *
+ * @param id the id of the post
+ * @param includeUser whether or not the user associated with the post should be returned in the data
+ * @returns the post with the given id
+ * @throws if the post with the given id cannot be found
+ */
+export async function getPost(
+  id: string,
+  includeUser: boolean
+): Promise<Post | PostWithUser> {
+  return prisma.post.findUnique({
+    where: {
+      id: id,
+    },
+    include: {
+      user: includeUser,
+    },
+    rejectOnNotFound: true,
```

Comment:
I think our convention has been to just return `null` if an entity is not found. I missed putting `null` as an output option in the spec though, could you fix that?

Reply:
Okay. I've added a "Null" type, and made it one of the possible outputs of the function.